### PR TITLE
Select datatype - Columnlength

### PR DESCRIPTION
**Pull request by ckogler-elements**
Made columnlength configurable
https://github.com/pimcore/pimcore/issues/2077

### CODE REVIEW

File: pimcore/models/Asset/Listing/Dao.php [hunk 1]
```diff
@@ -111,6 +111,8 @@ public function getTotalCount()
         $select = $this->getQuery([new Expression('COUNT(*)')]);
         $select->reset(QueryBuilder::LIMIT_COUNT);
         $select->reset(QueryBuilder::LIMIT_OFFSET);
+        $select->reset(QueryBuilder::ORDER);
```
Comment (weisswurstkanone) [inline pimcore/models/Asset/Listing/Dao.php]: how is this related to the whole task ?

Comment (ckogler-elements) [reply pimcore/models/Asset/Listing/Dao.php]: It was just a test if you review the code :p. 

The line is not really related to this Pull request. I think i had just selected the this branch when i committed the issue - sorry . 
Anyway - this should be merged in the core too.

**Comment by weisswurstkanone** [comment]
please resubmit the PR as there are conflicts.

**Comment by weisswurstkanone** [comment]
thx

**Comment by dpfaffenbauer** [comment]
This PR is actually kind of a BC break. Older Class-Definitions (prio to this change) can't now be used anymore. Since old definition file still have the queryColumnType and columnType like this:
```
"queryColumnType": "varchar(190)",
"columnType": "varchar(190)",
```

Pimcore now twice adds the length type:

```
ALTER TABLE `object_query_3` ADD COLUMN `orderLanguage` varchar(190)(190) NULL;
```

This does not only affect select types, it affects all subtypes of it. For example:

 - Language
 - Gender

see here for more: https://travis-ci.org/coreshop/CoreShop/jobs/298138353

In order to fix that, you should probably check if the columnType already has a bracket in it.

#2085 does the same for Pimcore 4.

**Comment by weisswurstkanone** [comment]
Hi,

this issue is taken into account by the update post-update script.
Only point here: the build number is wrong.
Fixed this via:
https://github.com/pimcore/pimcore/commit/2673a33900359eebd289bf0e7bbbb74a35a9ca4e

**Comment by dpfaffenbauer** [comment]
Hi,

But what about plugins/bundles installing classes? They now need to support two versions, the old one and the new one.

**Comment by ckogler-elements** [comment]
The old definition format is now supported too.

**Comment by dpfaffenbauer** [comment]
awesome, cheers (Y) :beers:
